web/src/app/procurement/pending/hooks.ts

```diff
@@ -1,70 +1,336 @@
+			fieldType: "button",
+			fieldProps: {
+				children: "Add Items",
+				name: "addItems",
+				color: "primary",
+				variant: "outlined",
+				sx: { width: 120 },
+				onClick: () => handleHeaderActions("addItems"),
+			},
+		},
+		{
+			fieldType: "button",
+			fieldProps: {
+				children: "Cancel Request",
+				name: "cancel",
+				color: "primary",
+				variant: "outlined",
+				sx: { width: 150 },
+				onClick: () => handleHeaderActions("cancel"),
+			},
 		},
 	];
 
-	const filteredDataSource = dataSource.filter(
-		(item) => item.status === "pending"
-	);
+	const handleAddItems = () => setOpen((state) => !state);
+
+	const handleSelectRow = (rows: RequestedDetailDto[]) => {
+		setSelectedRows(rows);
+	};
+
+	const handleHeaderActions = (type: ActionType) => {
+		switch (type) {
+			case "approve": {
+				if (!selectedRowData) {
+					console.warn("No row selected");
+					return;
+				}
+				console.log("Approve clicked");
+				break;
+			}
+			case "addItems":
+				handleAddItems();
+				break;
+			case "cancel":
+				console.log("Cancel Request clicked");
+				break;
+			default:
+				break;
+		}
+	};
+
+	const itemsOption: IOptionsProps[] = useMemo(() => {
+		if (!items) return [];
+		return transformObjectArrayToIOptionProps<ItemDto>({
+			data: items.data,
+			key: "id",
+			value: "item",
+		});
+	}, [items]);
+
+	const unitsOption: IOptionsProps[] = useMemo(() => {
+		if (!units) return [];
+		return transformObjectArrayToIOptionProps<UnitDto>({
+			data: units,
+			key: "id",
+			value: "units",
+		});
+	}, [units]);
+
+	const fields: IField<ICustomSelectProps | ICustomInputProps>[] = [
+		{
+			fieldType: "dropdown",
+			fieldProps: {
+				label: "Select items",
+				options: itemsOption,
+				name: "items",
+				control: control,
+				rules: { required: "Items is required" },
+			},
+		},
+		{
+			fieldType: "text",
+			fieldProps: {
+				label: "Quantity",
+				name: "quantity",
+				type: "text",
+				control: control,
+				rules: {
+					required: "Quantity is required",
+					pattern: {
+						value: /^(?:\d+|\d*\.\d+)$/,
+						message: "Only numbers are allowed",
+					},
+				},
+			},
+		},
+		{
+			fieldType: "dropdown",
+			fieldProps: {
+				label: "Select units",
+				options: unitsOption,
+				name: "units",
+				control: control,
+				rules: { required: "Units is required" },
+			},
+		},
+		{
+			fieldType: "text",
+			fieldProps: {
+				label: "Brand(optional)",
+				name: "brand",
+				control: control,
+			},
+		},
+		{
+			fieldType: "text",
+			fieldProps: {
+				label: "Specification",
+				name: "specification",
+				control: control,
+				rules: { required: "Input text is required" },
+				autoComplete: "off",
+			},
+		},
+	];
+
+	const filteredDataSource = dataSource.filter((item) => {
+		const status = item.status === ERequestStatus.PENDING;
+
+		const isAdmin = user?.access_data?.role_name === "Admin";
+		const department = isAdmin || user?.departments?.includes(item.department);
```

Comment:
e pasa ang department sa query params sa api
```suggestion
		const department =  user?.departments?.includes(item.department);
```